src/org/thoughtcrime/securesms/LogViewActivity.java

```diff
@@ -90,6 +97,28 @@ public boolean onOptionsItemSelected(MenuItem item) {
     return false;
   }
 
+  public void shareLog() {
+    try {
+      String timestamp = new SimpleDateFormat("yyyy-MM-dd-HH-mm").format(new Date());
+      File file = new File(getExternalCacheDir(), "log-" + timestamp + ".txt");
+      file.createNewFile();
+      file.setReadable(true, false);
+      FileWriter logFileWriter = new FileWriter(file, false);
+      BufferedWriter logFileBufferWriter = new BufferedWriter(logFileWriter);
+      logFileBufferWriter.write(logViewFragment.getLogText());
+      logFileBufferWriter.close();
```

Comment:
there is already LogViewFragment::saveLogFile() doing similar things, let's see to use the same code, to get also the same filenames etc.